netlink-packet-audit/src/codec.rs

```diff
@@ -0,0 +1,135 @@
+use std::{fmt::Debug, io};
+
+use bytes::BytesMut;
+use netlink_packet_core::{
+    NetlinkBuffer,
+    NetlinkDeserializable,
+    NetlinkMessage,
+    NetlinkSerializable,
+};
+pub(crate) use netlink_proto::{NetlinkCodec, NetlinkMessageCodec};
+
+/// audit specific implemenation of [`NetlinkMessageCodec`] due to the
+/// protocol violations in messages generated by kernal audit.
+///
+/// Among the known bugs in kernel audit messages:
+/// - `nlmsg_len` sometimes contains the padding too (it shouldn't)
+/// - `nlmsg_len` sometimes doesn't contain the header (it really should)
+///
+/// See also:
+/// - https://blog.des.no/2020/08/netlink-auditing-and-counting-bytes/
+/// - https://github.com/torvalds/linux/blob/b5013d084e03e82ceeab4db8ae8ceeaebe76b0eb/kernel/audit.c#L2386
+/// - https://github.com/mozilla/libaudit-go/issues/24
+/// - https://github.com/linux-audit/audit-userspace/issues/78
+pub struct NetlinkAuditCodec {
```

Comment:
I think this can just be

```
pub struct NetlinkAuditCodec;
```

Reply:
It would expose the type "constructor", without actually needing an instance of it ever.

But if you prefer a union type I can certainly change that (and probably for `NetlinkCodec` too?).

Reply:
Ah I see, this is great let's keep it as is.